llvm/lib/Target/AMDGPU/AMDGPUSplitModule.cpp

```diff
@@ -486,6 +486,27 @@ void SplitGraph::Node::visitAllDependencies(
   }
 }
 
+/// Checks if \p I has MD_callees and if it does, parse it and put the function
+/// in \p Callees.
+///
+/// \returns true if there was metadata and it was parsed correctly. false if
+/// there was no MD or if it contained unknown entries.
+static bool handleCalleesMD(const Instruction &I,
+                            SmallVector<Function *> &Callees) {
```

Comment:
I would expect the CallGraph analysis to parse this metadata for you. Also this would be a separate patch from "aliases and inline asm"

Reply:
I thought of this patch more as a general improvement on the CallsExternal case in the CG.
Is it okay if I retitle the patch, or do you want me to split this into 2 or 3 patches ?

Reply:
Split it